src/http-adapter/index.js

```diff
@@ -0,0 +1,203 @@
+/***
+ * HttpAdapter
+ * @module
+ */
+
+export const HTTP_METHOD = Object.freeze({
+  GET: 'GET',
+  PATCH: 'PATCH',
+  POST: 'POST',
+  DELETE: 'DELETE'
+})
+
+/**
+ * Adapter which works with data over HTTP, based on the options that
+ * are passed to it's constructor. This class is responsible for
+ * any CRUD operations that need to be carried out with your {@link https://jsonapi.org/ JSON:API} service
+ * from within the Backbone.Store.
+ * @param {Object} options -  An object that contains a property, `urlPrefix`,
+ * that defines the REST service that will be returning a {@link https://jsonapi.org/ JSON:API} response.
+ * The Object can also include a requestInterceptor/responseInterceptor functions to be used with internal requests
+ */
+export default class HttpAdapter {
+  constructor (options = {}) {
+    this.urlPrefix = options.urlPrefix
+    this.serializeRequests = false
+    this._outstandingRequests = new Set()
+    this.requestInterceptor = options.requestInterceptor || (() => {})
+    this.responseInterceptor = options.responseInterceptor || (() => {})
+  }
+
+  /**
+   * Build a url based off a JSON:API resource
+   * ex: buildUrl('cat', '6') with a urlPrefix of '/api' returns `/api/cat/6/`
+   * @param {string} type - the JSON:API type of the resource url being built
+   * @param {string} [id] - the JSON:API id, if any, of the resource url being built
+   * @returns {string} - the built url string relating to the JSON:API entity
+   */
+  buildUrl (type, id) {
+    let idPath = `/${id}`
+    let typePath = `/${type}`
+    let path = this.urlPrefix || ''
+
+    path += typePath
+
+    if (id != null) {
+      path += idPath
+    }
+
+    return `${path}/`
+  }
```

Comment:
Cleanup for a future PR:
```javascript
if (!type) throw `resource type is ${type}`;

let path = `${this.urlPrefix || ''}/${type}/;
if (id !== null && id !== undefined) {
  path += `${id}/`;
}
return path;
```

Reply:
I would feel confident implementing those changes here, but considering the scope I am not sure it is a good idea 😅